flink/flink-common/src/main/java/com/netease/arctic/flink/FlinkSchemaUtil.java

```diff
@@ -18,49 +18,100 @@
 
 package com.netease.arctic.flink;
 
+import static org.apache.flink.table.descriptors.DescriptorProperties.DATA_TYPE;
+import static org.apache.flink.table.descriptors.DescriptorProperties.EXPR;
+import static org.apache.flink.table.descriptors.DescriptorProperties.NAME;
+import static org.apache.flink.table.descriptors.DescriptorProperties.WATERMARK_ROWTIME;
+import static org.apache.flink.table.descriptors.DescriptorProperties.WATERMARK_STRATEGY_DATA_TYPE;
+import static org.apache.flink.table.descriptors.DescriptorProperties.WATERMARK_STRATEGY_EXPR;
+
 import com.netease.arctic.table.ArcticTable;
 import com.netease.arctic.table.PrimaryKeySpec;
 import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.apache.curator.shaded.com.google.common.collect.Maps;
 import org.apache.flink.table.api.TableColumn;
+import org.apache.flink.table.api.TableColumn.ComputedColumn;
 import org.apache.flink.table.api.TableSchema;
 import org.apache.flink.table.api.ValidationException;
 import org.apache.flink.table.api.WatermarkSpec;
+import org.apache.flink.table.types.DataType;
 import org.apache.flink.table.types.logical.LogicalType;
 import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.table.types.logical.utils.LogicalTypeParser;
 import org.apache.flink.table.types.utils.TypeConversions;
+import org.apache.iceberg.Schema;
 import org.apache.iceberg.types.Types;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 import java.util.function.Function;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /** An util that converts flink table schema. */
 public class FlinkSchemaUtil {
 
   private static final Logger LOG = LoggerFactory.getLogger(FlinkSchemaUtil.class);
+  public static final String FLINK_PREFIX = "flink";
+
+  public static final String COMPUTED_COLUMNS = "computed-column";
+
+  public static final String WATERMARK = "watermark";
 
   /**
-   * Convert a {@link RowType} to a {@link TableSchema}.
+   * Convert iceberg Schema to flink TableSchema.
    *
-   * @param rowType a RowType
-   * @param primaryKeys
+   * @param arcticSchema
+   * @param arcticProperties
    * @return Flink TableSchema
    */
-  public static TableSchema toSchema(RowType rowType, List<String> primaryKeys) {
+  public static TableSchema toSchema(
+      Schema arcticSchema, List<String> primaryKeys, Map<String, String> arcticProperties) {
```

Comment:
```suggestion
      Schema amoroSchema, List<String> primaryKeys, Map<String, String> amoroProperties) {
```
PLS also renames the word `arctic` to `amoro` in the comments.